core/src/from_meta.rs

```diff
@@ -274,6 +274,74 @@ impl<T: syn::parse::Parse, P: syn::parse::Parse> FromMeta for syn::punctuated::P
     }
 }
 
+/// Support for arbitrary expressions as values in a meta item.
+///
+/// For backwards-compatibility to versions of `darling` based on `syn` 1,
+/// string literals will be "unwrapped" and their contents will be parsed
+/// as an expression.
+impl FromMeta for syn::Expr {
+    fn from_expr(expr: &Expr) -> Result<Self> {
+        if let syn::Expr::Lit(expr_lit) = expr {
+            if let syn::Lit::Str(_) = &expr_lit.lit {
+                return Self::from_value(&expr_lit.lit);
```

Comment:
If `from_value` returns `Err` you could consider returning `Ok(expr.clone())` here too.